crates/driver/src/domain/competition/auction.rs

```diff
@@ -47,11 +43,18 @@ impl Auction {
         let weth = eth.contracts().weth_address();
         if !orders.iter().all(|order| {
             tokens.0.contains_key(&order.buy.token.wrap(weth))
-                && tokens.0.contains_key(&order.sell.token.wrap(weth))
+                && tokens.0.contains_key(&order.sell.token)
         }) {
             return Err(Error::InvalidTokens);
         }
 
+        // Ensure that there are no orders with 0 amounts.
+        if orders.iter().any(|order| {
+            order.solver_sell().amount.0.is_zero() || order.solver_buy(weth).amount.0.is_zero()
```

Comment:
Aren't 0 buy amounts for sell orders allowed?

Reply:
> Aren't 0 buy amounts for sell orders allowed?

AFAIK, we prohibit [0-amounts](https://github.com/cowprotocol/services/blob/4b68572058d00f82d15a8b015930af7d340e35d5/crates/shared/src/order_validation.rs#L629).